OverlayPlugin.Core/ClipboardTechSupport.cs

```diff
@@ -0,0 +1,156 @@
+    class ClipboardTechSupport
+    {
+        private SimpleTable plugins;
+        private SimpleTable overlays;
+        private SimpleTable settings;
+        private SimpleTable warnings;
+
+        public ClipboardTechSupport(TinyIoCContainer container)
+        {
+            var repository = container.Resolve<FFXIVRepository>();
+
+            warnings = new SimpleTable { new List<string> { "Warnings" } };
+            var opVersion = repository.GetOverlayPluginVersion();
+            if (opVersion < new Version(0, 19, 0))
+            {
+                warnings.Add(new List<string> { $"Out of date OverlayPlugin {opVersion.ToString()}, see https://overlayplugin.github.io/OverlayPlugin/fork_update.html" });
+            }
+
+
+            plugins = new SimpleTable { new List<string> { "Plugin Name", "Enabled", "Version", "Path" } };
+            foreach (var plugin in ActGlobals.oFormActMain.ActPlugins)
+            {
+                // TODO: plugin.pluginVersion has FileVersion, ProductVersion, etc, but all as a string.
+                // For now, ask FileVersionInfo to get this for us.
+                var fullPath = plugin.pluginFile.FullName;
+                var version = FileVersionInfo.GetVersionInfo(fullPath);
+                plugins.Add(new List<string>{
+                    plugin.pluginFile.Name,
+                    // TODO: could check plugin.pluginObj to see if it loaded successfully.
+                    plugin.cbEnabled.Checked.ToString(),
+                    version?.FileVersion?.ToString() ?? "",
+                    fullPath,
+                });
+            }
+
+            var pluginConfig = container.Resolve<IPluginConfig>();
+            overlays = new SimpleTable { new List<string> { "Overlay Name", "URL" } };
+            foreach (var overlay in pluginConfig.Overlays)
+            {
+                overlays.Add(new List<string>{
+                    overlay.Name,
+                    overlay.Url,
+                });
+            }
+
+            settings = new SimpleTable { new List<string> { "Various Settings", "Value" } };
+            settings.Add(new List<string> { "Plugin Language", repository.GetLanguage().ToString() });
+            settings.Add(new List<string> { "Machina Region", repository.GetMachinaRegion().ToString() });
+            // Only available when running the game.
+            string gameVersion = repository.GetGameVersion();
+            if (gameVersion != "")
+                settings.Add(new List<string> { "Game Version", gameVersion });
+
+            var cactbotConfig = GetCactbotConfig(pluginConfig);
+            if (cactbotConfig != null)
+            {
+                try
+                {
+                    var userDir = cactbotConfig["options"]["general"]["CactbotUserDirectory"];
+                    settings.Add(new List<string> { "Cactbot User Dir", userDir.ToString() });
+                }
+                catch { }
+            }
```

Comment:
I'd rather this be some generic code/logic that individual downstream plugins or overlays could extend, but I'm not sure exactly how that would look.

Reply:
I think generic code only makes sense if you expect this to grow at all or if you expect this to get out of date and break.  Are there even other downstream OP plugins other than cactbot?

My feeling here is that yes this is an abstraction violation, however it's also a one-off and so it's less complicated to just include it here for now.  It seems more likely that cactbot will merge into OP before this setting changes.  Or, it also seems more likely that somebody might refactor this code in this class (and now have to do it across multiple projects with staged releases) than for this setting itself to change.

I understand your sentiment, but I'd rather wait to see if other cases happen before creating plumbing and generic code here.

Reply:
Maybe a comment then to note that intention?

Reply:
Done.